src/cluster/placement/service/operator.go

```diff
@@ -0,0 +1,79 @@
+package service
+
+import (
+	"errors"
+
+	"github.com/m3db/m3/src/cluster/placement"
+)
+
+// NewPlacementOperator constructs a placement.Operator which performs transformations on the
+// given placement.
+// If initialPlacement is null, BuildInitialPlacement must be called before any operations on the
```

Comment:
s/null/nil|omitted

Reply:
alternatively, you could define this as 

`NewPlacementOperator(opts placement.Options, initPlacement ...placement.Placement)`

and then check len <= 1, etc

Reply:
up to you.